presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloClient.java

```diff
@@ -0,0 +1,1140 @@
+            while (tryAgain);
+        }
+        return mapping;
+    }
+
+    /**
+     * Creates the index tables from the given Accumulo table. No op if
+     * {@link AccumuloTable#isIndexed()} is false.
+     *
+     * @param table Table to create index tables
+     */
+    private void createIndexTables(AccumuloTable table)
+    {
+        try {
+            // If our table is indexed, create the index as well
+            if (table.isIndexed()) {
+                // Create index table and set the locality groups
+                Map<String, Set<Text>> indexGroups = Indexer.getLocalityGroups(table);
+                conn.tableOperations().create(table.getIndexTableName());
+                conn.tableOperations().setLocalityGroups(table.getIndexTableName(), indexGroups);
+
+                // Create index metrics table, attach iterators, and set locality groups
+                conn.tableOperations().create(table.getMetricsTableName());
+                conn.tableOperations().setLocalityGroups(table.getMetricsTableName(), indexGroups);
+                for (IteratorSetting s : Indexer.getMetricIterators(table)) {
+                    conn.tableOperations().attachIterator(table.getMetricsTableName(), s);
+                }
+            }
+        }
+        catch (Exception e) {
+            dropTable(table);
+            throw new PrestoException(INTERNAL_ERROR,
+                    "Accumulo error when creating index tables, check coordinator logs for more detail",
+                    e);
+        }
+    }
+
+    /**
+     * Drops the table metadata from Presto. If this table is internal, the Accumulo
+     * tables are also deleted.
+     *
+     * @param table The Accumulo table
+     */
+    public void dropTable(AccumuloTable table)
+    {
+        SchemaTableName stName = new SchemaTableName(table.getSchema(), table.getTable());
+        String tn = table.getFullTableName();
+
+        // Remove the table metadata from Presto
+        metaManager.deleteTableMetadata(stName);
+
+        // If the table is external, we won't clean up the Accumulo tables
+        if (table.isExternal()) {
+            return;
+        }
+
+        try {
+            // delete the table
+            conn.tableOperations().delete(tn);
+
+            // and delete our index tables
+            if (table.isIndexed()) {
+                conn.tableOperations().delete(Indexer.getIndexTableName(stName));
+                conn.tableOperations().delete(Indexer.getMetricsTableName(stName));
+            }
+        }
+        catch (Exception e) {
+            throw new PrestoException(INTERNAL_ERROR, "Failed to delete table",
+                    e);
+        }
+    }
+
+    /**
+     * Renames the metadata associated Accumulo table
+     *
+     * @param oldName Old table name
+     * @param newName New table name
+     */
+    public void renameTable(SchemaTableName oldName, SchemaTableName newName)
+    {
+        if (this.getTable(newName) != null) {
```

Comment:
It is not case insensitive (which I will add to my list to fix), but I am removing this and opting to use Accumulo APIs to test for table existence instead of Presto metadata.  Table rename operation would fail if a non-Presto Accumulo table exists.